### ISSUE TITLE
stream did not contain valid UTF-8

### ISSUE THREAD

**Issue by ghost** [herzrasen/pkghist#9]
Hi, as my Arch installation ages the pacman log is around 12MB and pkghist is a great tool for checking stuff. So I installed it from AUR but I'm getting a panic. Here's more info:

```
$ pacman -Q pkghist rust
pkghist 0.1.0-1
rust 1:1.37.0-2

$  file /var/log/pacman.log
/var/log/pacman.log: UTF-8 Unicode text, with very long lines, with escape sequences

$ pkghist --last 8
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" }', src/libcore/result.rs:999:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

$ RUST_BACKTRACE=1 pkghist --last 8
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" }', src/libcore/result.rs:999:5
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
10: <unknown>
11: <unknown>
12: <unknown>
13: <unknown>
14: <unknown>
15: __libc_start_main
16: <unknown>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

$ RUST_BACKTRACE=full pkghist --last 8
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" }', src/libcore/result.rs:999:5
stack backtrace:
   0:     0x55a4fde84b1b - <unknown>
   1:     0x55a4fde847f7 - <unknown>
   2:     0x55a4fde85230 - <unknown>
   3:     0x55a4fde84db2 - <unknown>
   4:     0x55a4fde84c96 - <unknown>
   5:     0x55a4fde97ddd - <unknown>
   6:     0x55a4fdd68fce - <unknown>
   7:     0x55a4fdd6fbb2 - <unknown>
   8:     0x55a4fdd83e0f - <unknown>
   9:     0x55a4fdd71887 - <unknown>
10:     0x55a4fdd6c930 - <unknown>
11:     0x55a4fde84c33 - <unknown>
12:     0x55a4fde86dfa - <unknown>
13:     0x55a4fde8573d - <unknown>
14:     0x55a4fdd719b2 - <unknown>
15:     0x7f056ef17ee3 - __libc_start_main
16:     0x55a4fdd620ae - <unknown>
17:                0x0 - <unknown>
```

If there's any additional info I can provide, feel free to ask.

**Comment by herzrasen** [comment]
Thanks for reporting this! 
Could you provide a .tar.gz of your pacman.log so I could debug into it?

**Comment by ghost** [comment]
@herzrasen Thanks for looking into this. Hopefully you can download my [pacman.log](https://we.tl/t-vZFKanbNJc) and debug. Much appreciated!

**Comment by herzrasen** [comment]
Thank you again. I was able to locate the problem. There are some weird control characters in the logfile. I guess it is most secure to ignore those lines completely when parsing them. I will release a new version tomorrow.

**Comment by herzrasen** [comment]
hi @glitsj16, 
I just released v0.1.1 via AUR, which fixes your problem. Hope you enjoy using pkghist.

**Comment by ghost** [comment]
@herzrasen Just did a few tests on the new build and everything is working as expected. I do enjoy using this, something that fits nicely into the Arch toolbox. A big thank you for looking into this so professionally!
